src/Test/WindowsFormsIntegration/FeatureTests/ElementHost/BubbledEvents/BubbledEvents.cs

```diff
@@ -0,0 +1,217 @@
+        _avTextbox.Background = System.Windows.Media.Brushes.Cyan;
+        _avStackPanel.Name = "avStackPanel";
+        _avStackPanel.Background = System.Windows.Media.Brushes.Cornsilk;
+        _avGrid.Name = "avGrid";
+        _avGrid.Background = System.Windows.Media.Brushes.Aqua;
+        _host.BackColor = System.Drawing.Color.Red;
+        _avTextbox.KeyDown += new System.Windows.Input.KeyEventHandler(avControl_KeyDown);
+        _avStackPanel.KeyDown += new System.Windows.Input.KeyEventHandler(avControl_KeyDown);
+        _avGrid.KeyDown += new System.Windows.Input.KeyEventHandler(avControl_KeyDown);
+        _avButton.Name = "avButton";
+        _avButton.KeyDown += new System.Windows.Input.KeyEventHandler(avControl_KeyDown);
+    }
+
+    protected override bool BeforeScenario(TParams p, System.Reflection.MethodInfo scenario)
+    {
+        _uiApp = UIObject.Root.Children.Find(UICondition.CreateFromName("BubbledEventsTest"));
+        this.Controls.Clear();
+        _host.Child = null;
+        _avStackPanel.Children.Clear();
+        _avGrid.Children.Clear();
+        _avDockPanel.Children.Clear();
+        _routingStrategy = string.Empty;
+        switch (scenario.Name)
+        {
+            case "Scenario1":
+                _avStackPanel.Children.Add(_avTextbox);
+                _host.Child = _avStackPanel;
+                this.Controls.Add(_host);
+                _sr = new ScenarioResult();
+                break;
+            case "Scenario2":
+                _avGrid.Children.Add(_avStackPanel);
+                _avStackPanel.Children.Add(_avTextbox);
+                _host.Child = _avGrid;
+                this.Controls.Add(_host);
+                _sr = new ScenarioResult();
+                break;
+            case "Scenario3":
+                _avGrid.Children.Add(_avStackPanel);
+                _avStackPanel.Children.Add(_avDockPanel);
+                _avDockPanel.Children.Add(_avTextbox);
+                _host.Child = _avGrid;
+                this.Controls.Add(_host);
+                _sr = new ScenarioResult();
+                break;
+            case "Scenario4":
+                _avGrid.Children.Add(_avStackPanel);
+                _avStackPanel.Children.Add(_avDockPanel);
+                _avDockPanel.Children.Add(_avButton);
+                _avButton.Content = _avTextbox;
+                _host.Child = _avGrid;
+                this.Controls.Add(_host);
+                _sr = new ScenarioResult();
+                break;
+        }
+        return base.BeforeScenario(p, scenario);
+    }
+
+    void avControl_KeyDown(object sender, System.Windows.Input.KeyEventArgs e)
+    {
+        //assign RoutingStrategy to global string so a given scenario can examine it.
+        //RoutingStrategy += ((System.Windows.FrameworkElement)sender).Name + ":" + e.RoutedEvent.RoutingStrategy.ToString() + "::";
+
+        //Changed 4/27/06 per Nathan Enright to only look at the routing strategy itself:
```

Comment:
@anjali-wpf Occurrence of date and name.

Reply:
Addressed